packages/babel-plugin-transform-regenerator/src/regenerator/emit.ts

```diff
@@ -56,1288 +30,1301 @@ const Ep = Emitter.prototype;
-// for assignments to temporary variables.
-Ep.emitAssign = function (lhs: any, rhs: any) {
-  this.emit(this.assign(lhs, rhs));
-  return lhs;
-};
+  // Sets the exact value of the given location to the offset of the next
+  // Statement emitted.
+  mark(loc: t.NumericLiteral) {
+    const index = this.listing.length;
+    if (loc.value === PENDING_LOCATION) {
+      loc.value = index;
+    } else {
+      // Locations can be marked redundantly, but their values cannot change
+      // once set the first time.
+      assert.strictEqual(loc.value, index);
+    }
+    this.marked[index] = true;
+    return loc;
+  }
 
-// Shorthand for an assignment statement.
-Ep.assign = function (lhs: any, rhs: any) {
-  const t = util.getTypes();
-  return t.expressionStatement(
-    t.assignmentExpression("=", t.cloneDeep(lhs), rhs),
-  );
-};
+  emit(node: t.Node) {
+    if (t.isExpression(node)) {
+      node = t.expressionStatement(node);
+    }
+    t.assertStatement(node);
+    this.listing.push(node);
+  }
 
-// Convenience function for generating expressions like context.next,
-// context.sent, and context.rval.
-Ep.contextProperty = function (name: any, computed: any) {
-  const t = util.getTypes();
-  return t.memberExpression(
-    this.getContextId(),
-    computed ? t.stringLiteral(name) : t.identifier(name),
-    !!computed,
-  );
-};
+  // Shorthand for emitting assignment statements. This will come in handy
+  // for assignments to temporary variables.
+  emitAssign<T extends t.AssignmentExpression["left"]>(
+    lhs: T,
+    rhs: t.Expression,
+  ) {
+    this.emit(this.assign(lhs, rhs));
+    return lhs;
+  }
 
-// Shorthand for setting context.rval and jumping to `context.stop()`.
-Ep.stop = function (rval: any) {
-  if (rval) {
-    this.setReturnValue(rval);
+  // Shorthand for an assignment statement.
+  assign(lhs: t.AssignmentExpression["left"], rhs: t.Expression) {
+    return t.expressionStatement(
+      t.assignmentExpression("=", t.cloneNode(lhs), rhs),
+    );
   }
 
-  this.jump(this.finalLoc);
-};
+  // Convenience function for generating expressions like context.next,
+  // context.sent, and context.rval.
+  contextProperty(name: string, computed?: boolean) {
+    return t.memberExpression(
+      this.getContextId(),
+      computed ? t.stringLiteral(name) : t.identifier(name),
+      !!computed,
+    );
+  }
 
-Ep.setReturnValue = function (valuePath: any) {
-  util.getTypes().assertExpression(valuePath.value);
+  // Shorthand for setting context.rval and jumping to `context.stop()`.
+  stop(rval: any) {
+    if (rval) {
+      this.setReturnValue(rval);
+    }
 
-  this.emitAssign(
-    this.contextProperty("rval"),
-    this.explodeExpression(valuePath),
-  );
-};
+    this.jump(this.finalLoc);
+  }
 
-Ep.clearPendingException = function (tryLoc: any, assignee: any) {
-  const t = util.getTypes();
+  setReturnValue(valuePath: any) {
+    t.assertExpression(valuePath.value);
```

Comment:
Is `valuePath` a `path`? If so it shouldn't have a `.value` property, it should be `.node` :thinking:

Reply:
It should be, it and `stop` seem to be dead code, maybe remove them?

Reply:
Oh yeah, it's unused